### PR TITLE
VueJS migration to typescript

### PR DESCRIPTION
**Pull request by NotExpectedYet**
I have migrated the previously working components and the whole vue base over to typescript. 

Happy for input tbh, as can clean it up in this PR for the current components. 

There's 1 issue I need to rectify and that's how to change the router options as it didn't like my use of icon and I have initially changed that inside the node_modules folder which obviously won't carry over. Be investigating shortly. 

Also changed the folder name to client_2x.

### CODE REVIEW

**Comment by NotExpectedYet** [comment]
Just to note it's these components I've done so far. 

Toolbar/Navigation 

The others are still normal vue javascript but will be working on those in another PR. I want to make a lot of changes to printer manager anyway.

**Comment by davidzwa** [comment]
Mate I am looking forward to checking this out!! 🥳

**Comment by NotExpectedYet** [comment]
I may end up adding some printer stuff tomorrow depends if I get time/this is commited aha. Was fun learning the type script world. I'll fix the Vue router issue though at the very least :)

File: client_2x/src/App.vue [hunk 1]
```diff
@@ -16,21 +16,27 @@
...
+export default class OctoFarm extends Vue {
+
+};
 </script>
+
 <style>
 main {
   background-color: black;
   background-image: linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)),
-    url("~@/assets/images/bg.jpg");
```
Comment (davidzwa) [inline client_2x/src/App.vue:33]: We have the paths entry in `tsconfig.json` for shorthand paths. Can look at it later.

File: client_2x/babel.config.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
```
Comment (davidzwa) [inline client_2x/babel.config.js:1]: https://github.com/studentenhuisDS4/dsnake4/blob/develop/dsnake4-phaser/babel.config.js

File: client_2x/.eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    "plugin:vue/essential",
+    "eslint:recommended",
+    "@vue/typescript/recommended",
+    "@vue/prettier",
+    "@vue/prettier/@typescript-eslint",
+  ],
+  parserOptions: {
+    ecmaVersion: 2020,
+  },
+  rules: {
```
Comment (davidzwa) [inline client_2x/.eslintrc.js:16]: https://github.com/studentenhuisDS4/dsnake4/blob/develop/dsnake4-phaser/.eslintrc.js

File: client_2x/src/App.vue [hunk 4]
```diff
@@ -16,21 +16,27 @@
   </v-app>
 </template>
 
-<script>
-import Vue from "vue";
+<script lang="ts">
```
Comment (davidzwa) [inline client_2x/src/App.vue:19]: No need: typescript never runs in the browser,

File: client_2x/src/store/index.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import Vue from "vue";
+import Vuex from "vuex";
+
+Vue.use(Vuex);
+
+export default new Vuex.Store({
```
Comment (davidzwa) [inline client_2x/src/store/index.ts:6]: Nice, we should use the store asap, stuff like stored filters (printer group dropdown) and server cache data.

File: client_2x/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    "skipLibCheck": true,
+    "esModuleInterop": true,
+    "allowSyntheticDefaultImports": true,
+    "sourceMap": true,
+    "baseUrl": ".",
+    "types": [
+      "webpack-env"
+    ],
+    "paths": {
+      "@/*": [
```
Comment (davidzwa) [inline client_2x/tsconfig.json:19]: Some magic with baseUrl and stuff should make this work. Yes bit finicky

**Comment by davidzwa** [comment]
https://vuejs.org/v2/guide/deployment.html#With-Build-Tools
Shall we hop onto webpack? I have really good experiences with it in production. It fits our idea of github release updates much better.

File: client_2x/.eslintrc.js (hunk 3 above)
Comment (NotExpectedYet) [reply client_2x/.eslintrc.js:16]: Vue typescript is deprecated as far as I'm aware.

**Comment by NotExpectedYet** [comment]
> 
> 
> https://vuejs.org/v2/guide/deployment.html#With-Build-Tools
> Shall we hop onto webpack? I have really good experiences with it in production. It fits our idea of github release updates much better.

Yeah agree'd will take a look at adding it.

File: client_2x/src/store/index.ts (hunk 5 above)
Comment (NotExpectedYet) [reply client_2x/src/store/index.ts:6]: Oh plan too! Printer Information, Settings, Permissions was what I thought off top of my head.

Comment (NotExpectedYet) [reply client_2x/src/store/index.ts:6]: Oh and notifications/alerts

File: client_2x/.eslintrc.js (hunk 3 above)
Comment (NotExpectedYet) [reply client_2x/.eslintrc.js:14]: Got rid of it, slow on the uptake this morning

**Review by davidzwa — APPROVED** [review]

**Comment by davidzwa** [comment]
image
Currently looking at why the icons dont load

**Comment by davidzwa** [comment]
image
Fixed, maybe grouping, (subtle/slight or edge) coloring or some separators can be nice. Nav is a bit busy to look at right now

**Comment by davidzwa** [comment]
@NotExpectedYet there's nothing stopping you from merging it right now. Webpack is already implemented by default, just checked. Any other stuff we will work on in follow-ups I presume! Nice work man, this code is feeling so comfortable already

**Comment by NotExpectedYet** [comment]
@davidzwa I agree there, I had some issues with child routes when I did that initially but certainly will work on it.

**Pull request merged**